### PR TITLE
[docs] - (generic) CPU Hog experiment

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: udit <uditgaurav@gmail.com>

- docs for CPU Hog generic experiment

### CODE REVIEW

**Review by amitbhatt818 — DISMISSED** [review]
/lgtm

**Review by amitbhatt818 — DISMISSED** [review]

**Review by imrajdas — DISMISSED** [review]
Looks good

File: docs/cpu-hog.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+id: cpu-hog
+title: CPU Hog Experiment Details
+sidebar_label: CPU Hog
```
Comment (ksatchit) [inline docs/cpu-hog.md:4]: See how we are formatting the side bar labels of other experiments. Can this be: `CPU Hog`

File: docs/cpu-hog.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+---
+------
+
+## Experiment Metadata
+
+| Type      | Description                         | Tested K8s Platform                                               |
+| ----------| ----------------------------------- | ------------------------------------------------------------------|
+| Generic   | Give a CPU spike on kubernetes Node | GKE, Minikube, OpenShift(Baremetal)                               |
```
Comment (ksatchit) [inline docs/cpu-hog.md]: Exhaust CPU resources on the Kubernetes Node

File: docs/cpu-hog.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+- Application pods are healthy on the respective Nodes before chaos injection
+
+## Exit Criteria
+
+- Application pods may or may not be healthy post chaos injection
+
+## Details
+
+- This experiment is used to inject a CPU spike chaos on a Kubernetes Node - which will help us to read the scenario where the node consumes the CPU equal to the CPU capacity to that node
```
Comment (ksatchit) [inline docs/cpu-hog.md]: This experiment causes CPU resource exhaustion on the Kubernetes node. The experiment aims to verify resiliency of applications whose replicas may be evicted on account on nodes turning unschedulable (`Not Ready`) due to lack of CPU resources.

File: docs/cpu-hog.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+- A stress tool is used to inject different kind of resource stress - such as CPU on the Kubernetes node. The stress command is executed in one of our daemonset pod (scheduled on the same node where the application pod is running) for the chaos duration
```
Comment (ksatchit) [inline docs/cpu-hog.md]: The CPU chaos is injected using a daemonset running the linux `stress` tool (a workload generator). The chaos is effected for a period equalling the TOTAL_CHAOS_DURATION.

File: docs/cpu-hog.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+- Tests the pod's behavior (application & services) running on the node which is under chaos experiment
```
Comment (ksatchit) [inline docs/cpu-hog.md]: application implies services. Can be reframed as: 

Tests application resiliency upon replica evictions caused due to lack of CPU resources

File: docs/cpu-hog.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+## Integrations
+
+- CPU Hog can be effected using one of these chaos libraries: `litmus`, `powerfulseal`
```
Comment (ksatchit) [inline docs/cpu-hog.md]: How powerfulseal?

File: docs/cpu-hog.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+- The desired chaos library can be selected by setting one of the above options as value for the env variable `LIB`
```
Comment (ksatchit) [inline docs/cpu-hog.md]: We don't have options for `LIB` other than `litmus`.

File: docs/cpu-hog.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+### Create the ChaosEngine Resource
+
+- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.
+
+  `kubectl apply -f chaosengine.yml`
+
+### Watch Chaos progress
+
+- View the CPU spike on the application node by setting up a watch on the node
```
Comment (ksatchit) [inline docs/cpu-hog.md]: setting up a watch of the CPU consumed by nodes in the Kubernetes Cluster

File: docs/cpu-hog.md (hunk 2 above)
Comment (ksatchit) [inline docs/cpu-hog.md]: Can we restrict this only to GKE? That is the only platform where self-recovery of nodes is possible. Also, we seem to use `watch kubectl top nodes` below. This is supported only if you have GKE/metric-server etc., I guess?

File: docs/cpu-hog.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+spec:
+  appinfo:
+    appns: default
+    applabel: 'app=nginx'
+    appkind: deployment
+  chaosServiceAccount: nginx-sa
+  monitoring: false
+  jobCleanUpPolicy: delete
+  experiments:
+    - name: pod-delete
```
Comment (ksatchit) [inline docs/cpu-hog.md]: We are doing CPU chaos right?  Not pod-delete ?

File: docs/cpu-hog.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+| Type      | Description                         | Tested K8s Platform                                               |
+| ----------| ----------------------------------- | ------------------------------------------------------------------|
+| Generic   | Give a CPU spike on kubernetes Node | GKE, Minikube, OpenShift(Baremetal)                               |
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running
+- Ensure that the `cpu-hog` experiment resource is available in the cluster. If not, install from [here](https://hub.litmuschaos.io/charts/generic/experiments/cpu-hog)
+
```
Comment (ksatchit) [inline docs/cpu-hog.md:19]: Can we add a note here saying there should be administrative access to the platform on which the Kubernetes cluster is hosted, as the recovery of the affected could be manual. Example `gcloud` access to the project.

File: docs/cpu-hog.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+- Provide the application info in `spec.appinfo`
+- Override the experiment tunables if desired 
+
+#### Supported Experiment Tunables
+
+| Variables             | Description                                                  | Type      | Notes                                                                             |
+| ----------------------| ------------------------------------------------------------ |-----------|------------------------------------------------------------|
+| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion (seconds)              | Optional  | Defaults to 60s                                                                   |
+| PLATFORM              | The platform on which the chaos experiment will run          | Optional  | Defaults to ANY                                                                   |
+| LIB                   | The chaos lib used to inject the chaos                       | Optional  | Defaults to `litmus`. Supported: `litmus`, `powerfulseal`                         |
```
Comment (ksatchit) [inline docs/cpu-hog.md]: Only litmus supported

File: docs/cpu-hog.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+- Provide the application info in `spec.appinfo`
+- Override the experiment tunables if desired 
+
+#### Supported Experiment Tunables
+
+| Variables             | Description                                                  | Type      | Notes                                                                             |
+| ----------------------| ------------------------------------------------------------ |-----------|------------------------------------------------------------|
+| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion (seconds)              | Optional  | Defaults to 60s                                                                   |
+| PLATFORM              | The platform on which the chaos experiment will run          | Optional  | Defaults to ANY                                                                   |
```
Comment (ksatchit) [inline docs/cpu-hog.md]: Lets make it GKE only.

**Review by ksatchit — CHANGES_REQUESTED** [review]
Have been busy with a bucket load of PRs today, hence the delay. Have requested changes on the basis of readability as well as usability. Please check.

File: docs/cpu-hog.md (hunk 2 above)
Comment (uditgaurav) [reply docs/cpu-hog.md]: true @ksatchit  done!

File: docs/cpu-hog.md (hunk 1 above)
Comment (uditgaurav) [reply docs/cpu-hog.md:4]: sidebar_labels for container kill and pod delete are not formatted and are written like this only. Do I need to change this?

**Review by ksatchit — APPROVED** [review]
Thanks! /lgtm

**Pull request merged**